docs/platform-integration/invoke-platform-code.md

```diff
@@ -0,0 +1,144 @@
+| -------- | ------ |
+| Android | _Platforms_ > _Android_ |
+| iOS | _Platforms_ > _iOS_ |
+| MacCatalyst | _Platforms_ > _MacCatalyst_ |
+| Windows | _Platforms_ > _Windows_ |
+
+> [!IMPORTANT]
+> Platform implementations must be in the same namespace and same class that the cross-platform API was defined in.
+
+The following screenshot shows the `DeviceOrientationService` classes in the _Android_ and _iOS_ folders:
+
+:::image type="content" source="media/invoke-platform-code/implement-api.png" alt-text="DeviceOrientationService classes in their Platforms folder screenshot.":::
+
+Alternatively, multi-targeting can be performed based on your own filename and folder criteria, rather than using the _Platforms_ folders. For more information, see [Configure multi-targeting](#configure-multi-targeting).
+
+### Android
+
+The following example shows the implementation of the `GetOrientation` method on Android:
+
+```csharp
+using Android.Content;
+using Android.Runtime;
+using Android.Views;
+
+namespace CallPlatformCodeDemos.Services
+{
+    public partial class DeviceOrientationService
+    {
+        public partial DeviceOrientation GetOrientation()
+        {
+            IWindowManager windowManager = Android.App.Application.Context.GetSystemService(Context.WindowService).JavaCast<IWindowManager>();
+            SurfaceOrientation orientation = windowManager.DefaultDisplay.Rotation;
+            bool isLandscape = orientation == SurfaceOrientation.Rotation90 || orientation == SurfaceOrientation.Rotation270;
+            return isLandscape ? DeviceOrientation.Landscape : DeviceOrientation.Portrait;
+        }
+    }
+}
+```
+
+### iOS
+
+The following example shows the implementation of the `GetOrientation` method on iOS:
+
+```csharp
+using UIKit;
+
+namespace CallPlatformCodeDemos.Services
+{
+    public partial class DeviceOrientationService
+    {
+        public partial DeviceOrientation GetOrientation()
+        {
+            UIInterfaceOrientation orientation = UIApplication.SharedApplication.StatusBarOrientation;
+            bool isPortrait = orientation == UIInterfaceOrientation.Portrait || orientation == UIInterfaceOrientation.PortraitUpsideDown;
+            return isPortrait ? DeviceOrientation.Portrait : DeviceOrientation.Landscape;
+        }
+    }
+}
+```
+
+## Invoke the cross-platform API
+
+After providing the platform implementations, the API can be invoked from cross-platform code by creating an object instance and invoking its operation:
+
+```csharp
+using CallPlatformCodeDemos.Services;
+using Microsoft.Maui.Controls;
+...
+
+DeviceOrientationService deviceOrientationService = new DeviceOrientationService();
+DeviceOrientation orientation = deviceOrientationService.GetOrientation();
+```
+
+At build time the build system will use multi-targeting to combine the cross-platform partial class with the partial class for the target platform, and build it into the app package.
+
+## Configure multi-targeting
```

Comment:
From this section I'm not really clear what I can use out of the box and what I need to configure if I want to use this. Maybe I need to click through to the other page, but maybe emphasize a bit more which method I can use without any additional configuration and which one is _also_ possible but requires a little work on my side.

Reply:
I thought you could use something out of the box, but it seems you can't :D